docs/reference/css_guidelines.rst

```diff
@@ -0,0 +1,151 @@
+-  Use ``//`` for comment blocks (instead of ``/* */``).
+-  Use single quotes for string values
+   ``background: url('my/image.png')`` or ``content: 'moose'``
+-  Avoid specifying units for zero values, e.g., ``margin: 0;`` instead
+   of ``margin: 0px;``.
+-  Strive to limit use of shorthand declarations to instances where you
+   must explicitly set all the available values.
+
+Sass imports
+~~~~~~~~~~~~
+
+Leave off underscores and file extensions in includes:
+
+.. code-block:: scss
+
+    // Bad
+    @import 'components/_widget.scss'
+
+    // Better
+    @import 'components/widget'
+
+Pixels vs. ems
+~~~~~~~~~~~~~~
+
+Use ``rems`` for ``font-size`` with a ``px`` fallback, because it offers
+absolute control over text. Additionally, unit-less ``line-height`` is
+preferred because it does not inherit a percentage value of its parent
+element, but instead is based on a multiplier of the ``font-size``.
+
+Specificity (classes vs. ids)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Prefer classes over IDs in CSS code. IDs can be overly specific and lead
+to duplication of CSS. When in doubt, use a class name.
+
+When styling a component, start with an element + class namespace,
+prefer direct descendant selectors by default, and use as little
+specificity as possible. Here is a good example:
+
+.. code-block:: html
+
+    <ul class="category-list">
+      <li class="item">Category 1</li>
+      <li class="item">Category 2</li>
+      <li class="item">Category 3</li>
+    </ul>
+
+.. code-block:: scss
+
+    .category-list { // element + class namespace
+
+      // Direct descendant selector > for list items
+      > li {
+        list-style-type: disc;
+      }
+
+      // Minimal specificity for all links
+      a {
+        color: #f00;
+      }
+    }
+
+Class naming conventions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Never reference ``js-`` prefixed class names from CSS files. ``js-`` are
+used exclusively from JS files.
+
+Use the SMACSS ``is-`` `prefix <https://smacss.com/book/type-state>`__
+for state rules that are shared between CSS and JS.
+
+Misc
+~~~~
+
+As a rule of thumb, avoid unnecessary nesting in SCSS. At most, aim for
+three levels. If you cannot help it, step back and rethink your overall
+strategy (either the specificity needed, or the layout of the nesting).
+
+Examples
+~~~~~~~~
+
+Here are some good examples that apply the above guidelines:
+
+.. code-block:: scss
+
+    // Example of good basic formatting practices
+    .styleguide-format {
+      color: #000;
+      background-color: rgba(0, 0, 0, .5);
+      border: 1px solid #0f0;
+    }
+
+    // Example of individual selectors getting their own lines (for error reporting)
+    .multiple,
+    .classes,
+    .get-new-lines {
+      display: block;
+    }
+
+    // Avoid unnecessary shorthand declarations
+    .not-so-good {
+      margin: 0 0 20px;
+    }
+    .good {
+      margin-bottom: 20px;
+    }
+
+Vendor prefixes
+~~~~~~~~~~~~~~~
+
+Where possible, use the autoprefixer
```

Comment:
Perhaps i've missed it, but I don't believe we're specifying an autoprefixer, making this a confusing reference.